### PR TITLE
[Ready to Merge]Toggle Karma

### PR DESCRIPTION
**Pull request by Couls**
**What does this PR do:**
Allows karma to be toggled, squashed version of #10972 

**Changelog:**
:cl:
add: Toggle karma button in special verbs
/:cl:

### CODE REVIEW

File: code/modules/client/preference/preferences_toggles.dm [hunk 1]
```diff
@@ -218,4 +218,18 @@
...
 		deactivate_darkmode()
-	feedback_add_details("admin_verb","TDarkmode") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
\ No newline at end of file
+	feedback_add_details("admin_verb","TDarkmode") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
+
+/client/verb/toggle_karma()
+	set name = "Toggle Karma Gains"
+	set category = "Special Verbs"
+	set desc = "This button will allow you to stop other people giving you karma."
+	if (prefs.toggles & DISABLE_KARMA)
```
Comment (Citinited) [inline code/modules/client/preference/preferences_toggles.dm]: Don't stick a space in `if (`

Comment (Couls) [reply code/modules/client/preference/preferences_toggles.dm]: nice catch, thanks

File: code/modules/client/client defines.dm [hunk 2]
```diff
@@ -60,6 +60,7 @@
 	var/karma = 0
 	var/karma_spent = 0
 	var/karma_tab = 0
+	var/wants_karma = TRUE // let's just keep it true as default for obvious reasons
```
Comment (Fox-McCloud) [inline code/modules/client/client defines.dm]: Why have this at all? You could just check for the pref when someone awards the karma, and if they have the pref, then it doesn't get awarded. This is kinda a redundant var.

Comment (Couls) [reply code/modules/client/client defines.dm]: its used for the proc can_get_karma which is inside karma.dm

File: code/modules/karma/karma.dm [hunk 3]
```diff
@@ -85,6 +85,9 @@ var/list/karma_spenders = list()
 		log_game("Illegal karma spending attempt detected from [key] to [M.key]. Using the same IP!")
 		to_chat(src, "<span class='warning'>You can't spend karma on someone connected from the same IP.</span>")
 		return FALSE
+	if(!M.client.wants_karma)
```
Comment (Fox-McCloud) [inline code/modules/karma/karma.dm]: why do it this way? You can just do

`if(M.get_preference(DISABLE_KARMA))`

File: code/modules/client/client defines.dm (hunk 2 above)
Comment (Fox-McCloud) [reply code/modules/client/client defines.dm]: I know, but why? Why have a separate var when you can check for the preference bitflag instead?

**Comment by Couls** [comment]
I honestly don't know why it merged a commit with nothing in it so weird

**Pull request merged**